src/main/scala/smile/modeling/AffineTransformation.scala

```diff
@@ -0,0 +1,138 @@
+    */
+
+  def forXAxisRelativeVerticalFlipOf(verticalSize: Double): AffineTransformation =
+    AffineTransformation(One, Zero, Zero, Zero, -One, -verticalSize)
+
+  /** Creates an affine transformation for a diagonal flip about the origin, flipping both
+    * horizontally and vertically.
+    *
+    * @param horizontalSize
+    *   The width of the image in pixels.
+    * @param verticalSize
+    *   The height of the image in pixels.
+    * @return
+    *   An affine transformation that flips the image diagonally.
+    */
+  def forOriginRelativeDiagonalFlipOf(
+      horizontalSize: Double,
+      verticalSize: Double
+  ): AffineTransformation =
+    AffineTransformation(-One, Zero, -horizontalSize, Zero, -One, -verticalSize)
+
+  /** Creates an affine transformation for rotation about a specified point.
+    *
+    * @param angle
+    *   The angle of rotation in degrees.
+    * @param pX
+    *   The X-coordinate of the point around which to rotate.
+    * @param pY
+    *   The Y-coordinate of the point around which to rotate.
+    * @return
+    *   An affine transformation representing the rotation.
+    */
+  def forPointCentredRotation(angle: Double, pX: Double, pY: Double): AffineTransformation =
+    val cos = MathUtils.cos(angle)
+    val sin = MathUtils.sin(angle)
+
+    forPointCentredRotation(cos, sin, pX, pY)
+
+  /** Creates an affine transformation for rotation about a specified point using precomputed cosine
+    * and sine values.
+    *
+    * @param cos
+    *   The cosine of the rotation angle.
+    * @param sin
+    *   The sine of the rotation angle.
+    * @param pX
+    *   The X-coordinate of the point around which to rotate.
+    * @param pY
+    *   The Y-coordinate of the point around which to rotate.
+    * @return
+    *   An affine transformation representing the rotation.
+    */
+  def forPointCentredRotation(
+      cos: Double,
+      sin: Double,
+      pX: Double,
+      pY: Double
+  ): AffineTransformation =
+    AffineTransformation(cos, sin, pX + sin * pY - cos * pX, -sin, cos, pY - sin * pX - cos * pY)
+
+/** Represents an affine transformation defined by matrix coefficients for scaling, rotating, and
+  * translating points in 2D space.
+  *
+  * @param alpha
+  *   The X scaling component.
+  * @param gamma
+  *   The X to Y shearing component.
+  * @param tauX
+  *   The X translation component.
+  * @param delta
+  *   The Y to X shearing component.
+  * @param beta
+  *   The Y scaling component.
+  * @param tauY
+  *   The Y translation component.
+  */
+case class AffineTransformation(
+    alpha: Double,
+    gamma: Double,
+    tauX: Double,
+    delta: Double,
+    beta: Double,
+    tauY: Double
+):
+  /** Converts this `AffineTransformation` to a Java AWT `AffineTransform`.
+    *
+    * @return
+    *   An `AWTAffineTransform` equivalent to this affine transformation.
+    */
+  def toAWTAffineTransform: AWTAffineTransform =
```

Comment:
There's only one place where this is called. Moving this method there would result this class becoming independent of AWT.